algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/molecular/GeneHasSpecificExonSkipping.kt

```diff
@@ -22,47 +22,61 @@ class GeneHasSpecificExonSkipping(override val gene: String, private val exonToS
             }
 
-            fusionSkippingEvents.isNotEmpty() -> {
+            exonSkippingFusions.isNotEmpty() -> {
                 EvaluationFactory.warn(
-                    "Exon $exonToSkip skipping in $gene (${concat(fusionSkippingEvents)}) " +
-                            "together with potentially exon skipping variant(s) (${concat(exonSplicingVariants)}",
-                    inclusionEvents = fusionSkippingEvents + exonSplicingVariants
+                    "$gene exon $exonToSkip skipping detected: ${concat(exonSkippingFusions)} " +
+                            "together with additional potentially exon $exonToSkip skipping variant(s) (${concat(exonSplicingVariants)}",
+                    inclusionEvents = exonSkippingFusions + exonSplicingVariants
                 )
             }
 
             exonSplicingVariants.isNotEmpty() -> {
                 EvaluationFactory.warn(
-                    "Potential $gene exon $exonToSkip skipping due to splice variant ${concat(exonSplicingVariants)}",
+                    "Potential $gene exon $exonToSkip skipping detected: ${concat(exonSplicingVariants)}",
                     inclusionEvents = exonSplicingVariants
                 )
             }
 
+            potentialExonSplicingVariants.isNotEmpty() -> {
+                EvaluationFactory.warn(
+                    "Potential $gene exon $exonToSkip skipping: variant(s) ${concat(potentialExonSplicingVariants)} detected in " +
+                            "splice region of exon $exonToSkip although unknown relevance (not annotated with splice coding effect)",
+                    inclusionEvents = potentialExonSplicingVariants
+                )
+            }
+
             else -> {
                 EvaluationFactory.fail("No $gene exon $exonToSkip skipping")
             }
         }
     }
 
-    private fun findExonSplicingVariants(molecular: MolecularTest) = molecular.drivers.variants.filter { variant ->
-        val isCanonicalExonAffected = variant.canonicalImpact.affectedExon != null && variant.canonicalImpact.affectedExon == exonToSkip
-        variant.isReportable && variant.gene == gene && isCanonicalExonAffected &&
-                (variant.canonicalImpact.codingEffect == CodingEffect.SPLICE || variant.canonicalImpact.isSpliceRegion == true)
-    }
-        .map(Variant::event)
-        .toSet()
+    private fun findExonSplicingVariants(molecular: MolecularTest, certainSplice: Boolean) =
+        molecular.drivers.variants.filter { variant ->
+            variant.isReportable && variant.gene == gene && variant.canonicalImpact.affectedExon != null
+                    && variant.canonicalImpact.affectedExon == exonToSkip
+                    && if (certainSplice) {
+                variant.canonicalImpact.codingEffect == CodingEffect.SPLICE
+            } else {
+                variant.canonicalImpact.isSpliceRegion == true
+            }
+        }
+            .map(Variant::event)
+            .toSet()
```

Comment:
I feel I asked this yesterday but is `affectedExon` populated correctly for splice region variants?

Also, I think I would make a separate function `isSplice(requireCertainty: Boolean)` to improve readability